src/coefficients/stdfunction_coefficient.hpp

```diff
@@ -0,0 +1,165 @@
+
+/**
+   \brief This method creates an array of size(local_elems), and assigns attributes based on the coefficient c
+
+   This method is useful for creating lists of attributes that correspond to elements in the mesh
+
+   \param[in] m The mesh
+   \param[inout] attr_list Should be an array that will hold the attributes that correspond to each element
+   \param[in] c The coefficient provided that will be evaluated on the mesh
+   \param[in] digitize An optional function that can be called to assign attributes based on the value of c at a given projection point. By default, values of c at a given d.o.f that are > 0. are assigned attribute 2, otherwise attribute 1.
+
+*/
+void MakeAttributeList(mfem::Mesh &m, mfem::Array<int> &attr_list,
+                       mfem::Coefficient &c,
+                       std::function<int(double)> = [](double v)
+{
+  return v > 0. ? 2 : 1;
+});
+
+/**
+   \brief This method creates an array of size(local_bdr_elems), and assigns attributes based on the coefficient c
+
+   This method is useful for creating lists of attributes that correspond to bdr elements in the mesh
+
+   \param[in] m The mesh
+   \param[inout] attr_list Should be an array that will hold the attributes that correspond to each element
+   \param[in] c The coefficient provided that will be evaluated on the mesh
+   \param[in] digitize An optional function that can be called to assign attributes based on the value of c at a given projection point. By default, values of c at a given d.o.f that are ==1. are assigned attribute 2, otherwise attribute 1. This means that only if all the d.o.f's of an bdr_element are "tagged" 1, will this bdr element be assigned attribute 2.
+
+*/
+void MakeBdrAttributeList(mfem::Mesh &m, mfem::Array<int> &attr_list,
+                          mfem::Coefficient &c,
+                          std::function<int(double)> = [](double v)
+{
+  return v == 1. ? 2 : 1;
+});
+
+/**
+   \brief AttributemodifierCoefficient class
+
+   This class temporarily changes the attribute to a given attribute list during evaluation
+*/
+class AttributeModifierCoefficient : public mfem::Coefficient
+{
+public:
+  /**
+     \brief This class temporarily changes the attribute during coefficient evaluation based on a given list.
+
+     \param[in] attr_list A list of attributes values corresponding to the type of coefficient at each element.
+     \param[in] c The coefficient to "modify" the element attributes
+  */
+  AttributeModifierCoefficient(const mfem::Array<int> &attr_list, mfem::Coefficient &c) :
+    m_attr_list(attr_list), m_C(&c) {}
+
+  virtual double Eval (mfem::ElementTransformation &Tr, const mfem::IntegrationPoint &ip);
+
+protected:
+  const mfem::Array<int> & m_attr_list;
+  mfem::Coefficient *m_C;
+};
+
+/**
+   TransformedVectorCoefficient applies various operations to modify a VectorCoefficient
+*/
+class TransformedVectorCoefficient : public mfem::VectorCoefficient
```

Comment:
Yeah, the more I'm looking at this, the more I think it should be separate files. That can be a follow-up change though.